utils/tests/test_system_check.py

```diff
@@ -0,0 +1,129 @@
+
+
+class SystemCheckTest(TestCase):
+    @override_settings(VALID_EMAIL_DOMAIN_LIST=["validdomain.it", "codepoets.it"])
+    def test_check_settings_valid_email_domain_list_with_correct_list(self):
+        errors = check_settings_valid_email_domain_list()
+
+        self.assertEqual(errors, [])
+
+    @override_settings(VALID_EMAIL_DOMAIN_LIST=["duplicate.com", "duplicate.com"])
+    def test_check_settings_valid_email_domain_list_warning_message_when_list_contains_duplicates(self):
+        errors = check_settings_valid_email_domain_list()
+
+        self.assertTrue(len(errors) > 0)
+        self.assertEqual(errors[0], create_warning_01_duplicated_values_in_valid_email_domain_list())
+
+    @override_settings(VALID_EMAIL_DOMAIN_LIST=666)
+    def test_check_settings_valid_email_domain_raise_error_when_it_is_not_instance_of_list(self):
+        errors = check_settings_valid_email_domain_list()
+
+        self.assertTrue(len(errors) > 0)
+        self.assertEqual(errors[0], create_error_01_valid_email_domain_list_is_not_a_list())
+
+    @override_settings(VALID_EMAIL_DOMAIN_LIST=[])
+    def test_check_settings_valid_email_domain_warning_message_when_list_is_empty(self):
+        errors = check_settings_valid_email_domain_list()
+
+        self.assertTrue(len(errors) > 0)
+        self.assertEqual(errors[0], create_warning_02_empty_valid_email_domain_list())
+
+    @override_settings(VALID_EMAIL_DOMAIN_LIST=["localhost"])
+    def test_check_settings_valid_email_domain_raise_error_when_domain_doesnt_have_dot(self):
+        errors = check_settings_valid_email_domain_list()
+        domain = settings.VALID_EMAIL_DOMAIN_LIST[0]
+
+        self.assertTrue(len(errors) > 0)
+        self.assertEqual(errors[0], create_error_02_invalid_domain_address(domain))
+
+    @override_settings(VALID_EMAIL_DOMAIN_LIST=["code/@,poets.it"])
+    def test_check_settings_valid_email_domain_raise_error_when_domain_contains_prohibited_signs(self):
+        errors = check_settings_valid_email_domain_list()
+        domain = settings.VALID_EMAIL_DOMAIN_LIST[0]
+
+        self.assertTrue(len(errors) > 0)
+        self.assertEqual(errors[0], create_error_02_invalid_domain_address(domain))
+
+    @override_settings()
+    def test_check_settings_valid_email_domain_raise_error_when_settings_doesnt_exist(self):
+        del settings.VALID_EMAIL_DOMAIN_LIST
+        with self.assertRaises(AttributeError):
+            check_settings_valid_email_domain_list()
+
+    @override_settings(SITE_ID="STRING")
+    def test_check_settings_site_id_raise_error_when_it_is_not_instance_of_integer(self):
+        errors = check_settings_site_id()
+
+        self.assertTrue(len(errors) > 0)
+        self.assertEqual(errors[0], create_error_04_site_id_value_must_be_integer())
+
+    @override_settings()
+    def test_check_settings_site_id_raise_error_when_settings_doesnt_exist(self):
+        del settings.SITE_ID
+        with self.assertRaises(AttributeError):
```

Comment:
this is wrong. Please write code that would check if `SITE_ID` exist in settings and returns custom `Error` if not